code_search/pipeline/index_update_pipeline.py

```diff
@@ -0,0 +1,127 @@
+
+    With this service account, the container has a range of GCP APIs to
+    access to. This service account is automatically created as part of the
+    kubeflow deployment.
+
+    For the list of the GCP APIs this service account can access to, check
+    https://github.com/kubeflow/kubeflow/blob/7b0db0d92d65c0746ac52b000cbc290dac7c62b1/deployment/gke/deployment_manager_configs/iam_bindings_template.yaml#L18
+
+    If you want to call the GCP APIs in a different project, grant the kf-user
+    service account access permission.
+  """
+
+  return (
+      dsl.ContainerOp(
+          name,
+          image,
+          command,
+          arguments,
+          file_inputs,
+          file_outputs,
+          is_exit_handler,
+      )
+      .add_volume(
+          k8s_client.V1Volume(
+              name='gcp-credentials',
+              secret=k8s_client.V1SecretVolumeSource(
+                  secret_name='user-gcp-sa'
+              )
+          )
+      )
+      .add_volume_mount(
+          k8s_client.V1VolumeMount(
+              mount_path='/secret/gcp-credentials',
+              name='gcp-credentials',
+          )
+      )
+      .add_env_variable(
+          k8s_client.V1EnvVar(
+              name='GOOGLE_APPLICATION_CREDENTIALS',
+              value='/secret/gcp-credentials/user-gcp-sa.json'
+          )
+      )
+  )
+
+def dataflow_function_embedding_op(project: 'GcpProject', runner: str, target_dataset: str, problem: str,
+                           data_dir: 'GcsUri',saved_model_dir: 'GcsUri',temp_location: 'GcsUri', staging_location: 'GcsUri',
+                           job_name: str, worker_machine_type: str,
+                           num_workers: int, step_name='dataflow_function_embedding'):
+    return default_gcp_op(
+        name = step_name,
+        image = 'gcr.io/yang-codesearch/code-search-dataflow:v20181124-23deb39',
+        command = [
+            'python2',
+            '-m',
+            'code_search.dataflow.cli.create_function_embeddings',
+        ],
+        arguments = [
+            '--project', project,
+            '--runner', runner,
+            '--target_dataset', target_dataset,
+            '--problem', problem,
+            '--data_dir', data_dir,
+            '--saved_model_dir', saved_model_dir,
+            '--job_name', job_name,
+            '--temp_location', temp_location,
+            '--staging_location', staging_location,
+            '--worker_machine_type', worker_machine_type,
+            '--num_workers', num_workers,
+            '--requirements_file', 'requirements.dataflow.txt',
+            '--wait_until_finished',
+        ]
+    )
+
+def search_index_creator_op(working_dir: str, data_dir: str, component: str, cluster_name: str, workflow_id: str):
+    return dsl.ContainerOp(
+        # use component name as step name
+        name = component,
+        image = 'gcr.io/yang-codesearch/code-search-search-index-creator:v20181126-6a63824',
```

Comment:
Same comment as above; can we reuse the Docker images in kubeflow-examples that the individual steps are already using?

Reply:
i've updated the dataflow image to reuse the existing one.